server/src/test/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapMasterIT.java

```diff
@@ -0,0 +1,236 @@
+            command.execute(terminal, options, environment);
+            assertThat(terminal.getOutput(), containsString(UnsafeBootstrapMasterCommand.MASTER_NODE_BOOTSTRAPPED_MSG));
+        } finally {
+            assertThat(terminal.getOutput(), containsString(UnsafeBootstrapMasterCommand.WARNING_MSG));
+        }
+    }
+
+    private void executeCommand(Environment environment) throws Exception {
+        executeCommand(environment, false);
+    }
+
+    private void expectThrows(ThrowingRunnable runnable, String message) {
+        ElasticsearchException ex = expectThrows(ElasticsearchException.class, runnable);
+        assertThat(ex.getMessage(), containsString(message));
+    }
+
+    public void testAbortedByUser() {
+        final Environment environment = TestEnvironment.newEnvironment(internalCluster().getDefaultSettings());
+        expectThrows(() -> executeCommand(environment, true), UnsafeBootstrapMasterCommand.ABORTED_BY_USER_MSG);
+    }
+
+    public void testNotMasterEligible() {
+        final Environment environment = TestEnvironment.newEnvironment(Settings.builder()
+                .put(internalCluster().getDefaultSettings())
+                .put(Node.NODE_MASTER_SETTING.getKey(), false)
+                .build());
+        expectThrows(() -> executeCommand(environment), UnsafeBootstrapMasterCommand.NOT_MASTER_NODE_MSG);
+    }
+
+    public void testNoDataFolder() {
+        final Environment environment = TestEnvironment.newEnvironment(internalCluster().getDefaultSettings());
+        expectThrows(() -> executeCommand(environment), UnsafeBootstrapMasterCommand.NO_NODE_FOLDER_FOUND_MSG);
+    }
+
+    public void testNodeLocked() throws IOException {
+        Settings envSettings = buildEnvSettings(Settings.EMPTY);
+        Environment environment = TestEnvironment.newEnvironment(envSettings);
+        try (NodeEnvironment ignored = new NodeEnvironment(envSettings, environment)) {
+            expectThrows(() -> executeCommand(environment), UnsafeBootstrapMasterCommand.FAILED_TO_OBTAIN_NODE_LOCK_MSG);
+        }
+    }
+
+    public void testNoNodeMetaData() throws IOException {
+        Settings envSettings = buildEnvSettings(Settings.EMPTY);
+        Environment environment = TestEnvironment.newEnvironment(envSettings);
+        try (NodeEnvironment nodeEnvironment = new NodeEnvironment(envSettings, environment)) {
+            NodeMetaData.FORMAT.cleanupOldFiles(-1, nodeEnvironment.nodeDataPaths());
+        }
+
+        expectThrows(() -> executeCommand(environment), UnsafeBootstrapMasterCommand.NO_NODE_METADATA_FOUND_MSG);
+    }
+
+    public void testNotBootstrappedCluster() throws Exception {
+        internalCluster().startNode(
+                Settings.builder()
+                        .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), Integer.MAX_VALUE)
+                        .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "2s") //to ensure quick node startup
```

Comment:
Why not zero?

Reply:
https://github.com/elastic/elasticsearch/pull/37696/commits/6552d541f5eb6bacee120200b4835a2c1cd4da4d